### PR TITLE
Fixes the multinomial grad dtype bug

### PR DESCRIPTION
**Pull request by caglar**

### CODE REVIEW

File: theano/sandbox/multinomial.py [hunk 1]
```diff
@@ -52,7 +52,11 @@ def make_node(self, pvals, unis, n=1):
     def grad(self, ins, outgrads):
         pvals, unis, n = ins
         (gz,) = outgrads
-        return [T.zeros_like(x) for x in ins]
+        if x.dtype in T.discrete_dtypes:
```
Comment (nouiz) [inline theano/sandbox/multinomial.py]: You can't do it like this. x isn't defined here. You need the if to be inside the loop. Each component much be checked individually.

Comment (caglar) [inline theano/sandbox/multinomial.py]: Opps sorry pushed the wrong version by mistake!

**Comment by caglar** [comment]
@nouiz @lamblin seems like tests are passing.

**Comment by nouiz** [comment]
thanks

**Pull request merged**